operator-framework/src/main/java/io/javaoperatorsdk/operator/api/RetryInfo.java

```diff
@@ -2,19 +2,24 @@
 
 public class RetryInfo {
```

Comment:
What about turning it into an interface extended by **RetryExecution** and implemented by **GenericRetryExecution**?
so the execution can be returned as info and no need to instantiate a new RetryInfo on every call.

Reply:
Hehe, I like to work with Pojos, but will check how that would look like, I change it in case its nice.